lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -171,6 +171,12 @@ sub create_from_settings {
         }
     }
 
+    for my $key (keys %settings) {
+        my $value = $settings{$key};
+        $settings{$key} = encode_json($value) if (ref $value eq 'ARRAY' || ref $value eq 'HASH');
```

Comment:
Would you mind adding some test coverage? The fact that this doesn't break anything suggests this is untested 😶